ignite/metrics/gan/fid.py

```diff
@@ -137,21 +120,19 @@ def __init__(
             raise RuntimeError("This module requires scipy to be installed.")
 
         # default is inception
-        if num_features is None and feature_extractor is None:
-            num_features = 2048
-            feature_extractor = InceptionExtractor()
-        elif num_features is None:
-            raise ValueError("Argument num_features should be defined")
-        elif feature_extractor is None:
-            self._feature_extractor = lambda x: x
-            feature_extractor = self._feature_extractor
-
-        if num_features <= 0:
-            raise ValueError(f"Argument num_features must be greater to zero, got: {num_features}")
-        self._num_features = num_features
-        self._feature_extractor = feature_extractor
+        self._default_channels = 2048
+        self._default_eval_model = InceptionModel
+        self._default_args = True
+
+        self._num_features, self._feature_extractor = self._check_input(num_channels, evaluation_model, device)
```

Comment:
Why you still call it `_feature_extractor`, please rename for consistency ?

Reply:
I agree, extractor was relevant for FID but not anymore.